main.go

```diff
@@ -140,21 +124,38 @@ func main() {
 				"GCP-ZONE":    zone,
 				"INSTANCE-IP": ip,
 			}
+		default:
+			fmt.Fprintf(os.Stderr, "Error: unknown deployment type \n")
+			os.Exit(1)
 		}
 	}
 
-	var meshId string
-	if *generateMeshId {
+	meshId := *configMesh
+	if meshId == "" && *generateMeshId {
```

Comment:
There should be an error if both `--config-mesh` and `--generate-mesh-id` are supplied, right?

Reply:
if both `--config-mesh` and `--generate-mesh-id` are supplied, the former will have precedence and will be used as meshID

Reply:
This seems counterintuiive, no? Why not simply reject this case?

Reply:
Makes sense. Done.